models/apple_store__territory_report.sql

```diff
@@ -1,95 +1,98 @@
-        and reporting_grain.app_id = app_store_territory.app_id 
-        and reporting_grain.source_type = app_store_territory.source_type
-        and reporting_grain.territory = app_store_territory.territory
-    left join downloads_territory
-        on reporting_grain.date_day = downloads_territory.date_day
-        and reporting_grain.source_relation = downloads_territory.source_relation
-        and reporting_grain.app_id = downloads_territory.app_id 
-        and reporting_grain.source_type = downloads_territory.source_type
-        and reporting_grain.territory = downloads_territory.territory
-    left join usage_territory
-        on reporting_grain.date_day = usage_territory.date_day
-        and reporting_grain.source_relation = usage_territory.source_relation
-        and reporting_grain.app_id = usage_territory.app_id 
-        and reporting_grain.source_type = usage_territory.source_type
-        and reporting_grain.territory = usage_territory.territory
+        coalesce(ip.impressions, 0) as impressions,
+        coalesce(ip.impressions_unique_device, 0) as impressions_unique_device,
+        coalesce(ip.page_views, 0) as page_views,
+        coalesce(ip.page_views_unique_device, 0) as page_views_unique_device,
+        coalesce(dd.first_time_downloads, 0) as first_time_downloads,
+        coalesce(dd.redownloads, 0) as redownloads,
+        coalesce(dd.total_downloads, 0) as total_downloads,
+        coalesce(sa.active_devices, 0) as active_devices,
+        coalesce(id.deletions, 0) as deletions,
+        coalesce(id.installations, 0) as installations,
+        coalesce(sa.sessions, 0) as sessions
+    from reporting_grain as rg
+    left join app as a
+        on rg.app_id = a.app_id
+        and rg.source_relation = a.source_relation
+    left join impressions_and_page_views as ip 
+        on rg.app_id = ip.app_id
+        and rg.date_day = ip.date_day
+        and rg.source_type = ip.source_type
+        and rg.territory = ip.territory
+        and rg.source_relation = ip.source_relation
+    left join downloads_daily as dd
+        on rg.app_id = dd.app_id
+        and rg.date_day = dd.date_day
+        and rg.source_type = dd.source_type
+        and rg.territory = dd.territory
+        and rg.source_relation = dd.source_relation
+    left join install_deletions as id
+        on rg.app_id = id.app_id
+        and rg.date_day = id.date_day
+        and rg.source_type = id.source_type
+        and rg.territory = id.territory
+        and rg.source_relation = id.source_relation
+    left join sessions_activity as sa
+        on rg.app_id = sa.app_id
+        and rg.date_day = sa.date_day
+        and rg.source_type = sa.source_type
+        and rg.territory = sa.territory
+        and rg.source_relation = sa.source_relation
     left join country_codes as official_country_codes
-        on reporting_grain.territory = official_country_codes.country_name
+        on rg.territory = official_country_codes.country_name
```

Comment:
Using our seed data it seems this needs to be changed as the seed data reflects this is now looking like territory_short as opposed to country_name. Can you confirm that in the live data this is accurate?
```suggestion
        on rg.territory = official_country_codes.country_code_alpha_2
```

Reply:
This is correct! Updated to using country_code_alpha_2 since territory is in short-form.